### PR TITLE
Update status when removing volunteer + use API data for table

### LINKED ISSUE #250: Trigger remove volunteer modal in admin schedule review table + update status
Open the modal in #159 when the remove button of the table row from #156 is clicked. 

When a volunteer is removed, call the `updateShiftSignup` mutation to change the status of the signup to `CANCELLED`.

### PR DESCRIPTION
**Pull request by lambo-liu**
## Ticket link

Closes #250 

## Implementation description
* Implemented the API call for the remove volunteer modal
* Connected table data with results from GraphQL API call
* Fixed minor bugs in the volunteer table, such as duplicate component keys.

## Steps to test
1. Navigate to `http://localhost:3000/admin/schedule/posting/{id}` 
2. Click the "Review" button to go to the review table
3. Check that the table is being filled with the correct data.
4. Try removing a signup.
5. Check that the signup has been removed from the table
6. Refresh the page and go to the side panel. The signup should now be uncheckmarked.

## What should reviewers focus on?
* Correct implementation & functionality

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/admin/schedule/AdminScheduleTableRow.tsx [hunk 1]
```diff
@@ -1,46 +1,76 @@
...
-        </Td>
-      </Tr>
+      <>
+        {volunteer && numVolunteers && note && (
+          <RemoveVolunteerModal
+            name={volunteer.name}
+            isOpen={isModalOpen}
+            shiftId={shiftId}
+            userId={volunteer.userId}
+            status="CANCELED"
```
Comment (LenaNguyen) [inline frontend/src/components/admin/schedule/AdminScheduleTableRow.tsx:39]: PENDING

**Review by LenaNguyen — CHANGES_REQUESTED** [review]
I'm using the seed data rn and there seems to be a bug where the remove modal does not pop up for all rows??? It looks like it might be every other row

**Review by LenaNguyen — COMMENTED** [review]
Also, in a different pr, we'll need to update the logic of the table rows a bit. We want to render a row for every day of the week. If there are no shifts for that day then we display "No Shifts Available". If there are no volunteers for the shifts, then we display the shift time followed by "No volunteers".

Designs:
image


Current: 
image

**Review by LenaNguyen — COMMENTED** [review]
I addressed the pr comments but will need someone else to review/approve now!

**Review by MatoPlus — DISMISSED** [review]
[last commit](https://github.com/uwblueprint/sistering/pull/304/commits/f487021491c32a6425c02f8df16ce7ca526d0d96) LGTM

**Review by LenaNguyen — APPROVED** [review]

**Review by LenaNguyen — APPROVED** [review]

**Review by LenaNguyen — APPROVED** [review]

**Pull request merged**